src/java/org/apache/cassandra/db/marshal/SetType.java

```diff
@@ -69,6 +70,17 @@ public SetType(AbstractType<T> elements, boolean isMultiCell)
         this.isMultiCell = isMultiCell;
     }
 
+    @Override
+    public SetType<T> overrideKeyspace(Function<String, String> overrideKeyspace)
+    {
+        List<AbstractType<?>> subTypes = subTypes();
+        AbstractType<?> newType = subTypes.get(0).overrideKeyspace(overrideKeyspace);
+        if (newType == subTypes.get(0))
+            return this;
+
+        return new SetType<>((AbstractType<T>) newType, isMultiCell());
```

Comment:
```suggestion
        return getInstance((AbstractType<T>) newType, isMultiCell());
```
`getInstance` will use the cached `SetType` to reduce GC caused by expensive `SetSerializer`.